### ISSUE TITLE
Suggestion: warn user before rendering N rows

### ISSUE THREAD

**Issue by daattali** [rstudio/gt#210]
Just an idea. I just tried running `gt::gt(gt::pizzaplace)` in the rstudio console and I thought my RStudio got frozen. I tried it several times, and each time I had to abort the operation. Eventually I realized it just took a full minute to complete the render.

Perhaps it'd be friendly to the user to show a message confirming that the user indeed wants to print the entire table if gt notices that there are above N number of rows? Of course only do this in interactive mode. Or maybe even just showing a message would be enough.

**Comment by rich-iannone** [comment]
I shot myself in the foot with this probably five times. Seemingly each time I thought there was a system issue and didn't realize it was a huge render. 

Basically, this is a good idea. Need to think about the right implementation (but there are lots of case studies for this sort of thing).

**Comment by sharlagelfand** [comment]
I personally like `visdat::vis_dat()`'s approach, which provides a warning if your data is above a certain size (ncol*nrow > 900000), but you can turn off this warning using the `warn_large_data` parameter (see [here](https://github.com/ropensci/visdat/blob/1c3cdbe19a49792c414bec9cf9d35b7c90653cf8/R/vis-dat.R#L20))

**Comment by jcheng5** [comment]
Hi @daattali, I didn't realize how slow `gt::gt(gt::pizzaplace)` was, so thanks even for that feedback alone!

@rich-iannone and I talked about this in person. I'd like to investigate two other paths first:

1. It's been a while since we've done an optimization pass on gt, a cursory investigation last week showed that huge gains are easily attainable.
2. Showing a progress bar for larger tables/slower renders would convey all of the same information and more. This is assuming that your R session doesn't have any difficulty cancelling a slower render? (Worked fine for us in RStudio when we tried it)
